query-planner-js/src/buildPlan.ts

```diff
@@ -1138,13 +1174,49 @@ class FetchGroup {
     const inputNodes = inputs ? inputs.toSelectionSetNode() : undefined;
 
     const subgraphSchema = this.dependencyGraph.subgraphSchemas.get(this.subgraphName)!;
-    let operation = this.isEntityFetch
-      ? operationForEntitiesFetch(
-          subgraphSchema,
-          this.selection,
-          variableDefinitions,
+    let operation: Operation;
+    const finders = this.isEntityFetch ? this.getFindersForInput(inputs) : [];
+
+    if (finders.length > 0) {
+      // if multiple finders are found, just use the first one for now
+      const finder = finders[0];
+      assert(finder, 'operationForFindersFetch called without a finder');
+      const operation = operationForFindersFetch({
+        subgraphSchema: subgraphSchema,
+        selectionSet: this.selection,
+        allVariableDefinitions: variableDefinitions,
+        operationName,
+        finder,
+      });
+      const operationDocument = operationToDocument(operation);
+      const subgraphFetchNode: SubgraphFetchNode = {
+        kind: 'SubgraphFetch',
+        serviceName: this.subgraphName,
+        id: this.id,
+        operationName: operation.name ?? 'finder call',
```

Comment:
This defaulting of the operation name is what currently breaks any query involving finders if the user didn't explicitly named the query. The reason being that, if that's the case, we indicate here that the operation is called "finder call", but we don't actually name it that way later in `operationForFindersFetch`, so the execution fails with 'Unknown operation named "finder call".'. Note that even if we were to name the operation similarly, "finder call" is not a valid operation name in graphQL.

And taking a step back, what's the reasoning for setting a default operation name? We didn't do it before for `_entities` query. I'd personally not add one here too.